app/src/main/java/com/codesoom/assignment/application/UserServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.User;
+import com.codesoom.assignment.domain.UserRepository;
+import com.codesoom.assignment.dto.UserData;
+import com.codesoom.assignment.dto.UserEmailDuplicateException;
+import com.codesoom.assignment.dto.UserNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Transactional
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+    public UserServiceImpl(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public User createUser(UserData source) throws Exception {
+
+        if( emailCheck(source.getEmail()) ) {
+            throw new UserEmailDuplicateException();
+        }
+
+        User user = User.builder()
+                .name(source.getName())
+                .email(source.getEmail())
+                .password(source.getPassword())
+                .build();
+
+        return userRepository.save(user);
+
+    }
+
+    @Override
+    public User updateUser(Long id, UserData source) {
+
+        User findUser = userRepository.findById(id).orElseThrow(() -> new UserNotFoundException());
+        findUser.userUpdate(source.getName(), source.getEmail(), source.getPassword());
+
+        return findUser;
+
+    }
+
+    @Override
+    public void deleteUser(Long id) {
+
+        userRepository.deleteById(id);
+
+    }
+
+    @Override
+    public boolean emailCheck(String mail) {
+
+        List<User> users = userRepository.findAll();
+
+        for (User user : users) {
+
+            if(user.getEmail().equals(mail)) {
+                return true;
+            }
+
+        }
+
+        return false;
+
+    }
```

Comment:
코드 사이사이에 빈 줄이 너무 많아서, 어떤 코드들이 같이 읽어야 하고 따로 일거야 하는지 잘 모르겠습니다.
```suggestion
    @Override
    public boolean emailCheck(String mail) {
        List<User> users = userRepository.findAll();

        for (User user : users) {
            if(user.getEmail().equals(mail)) {
                return true;
            }
        }

        return false;
    }
```

저는 이런 모양이었으면 좋겠어요

Reply:
너무 많은 개행도 가독성을 떨어뜨릴 수가 있군요..! 앞으로 개행을 적절히 하겠습니다 !